src/compiler/crystal/syntax/lexer.cr

```diff
@@ -1825,7 +1824,7 @@ module Crystal
       end
 
       # 0o177777_77777777_77777777 is the largest UInt64.
-      num = nil if num_size > 22 # or > 21 with first digit being 2 through 7
+      num = nil if {num_size, first_digit} > {22, 0o1}
```

Comment:
This condition looks unconventional and I find it a little harder to read. Maybe this would be a better way to write this?
```suggestion
      num = nil if num_size > 22 || (num_size == 22 && first_digit == 1)
```
It's fine by me to keep it that way, though.

Reply:
Well because it's the latest idea of mine, obviously I prefer to keep it 😀
Not a big deal though, others can weigh in as well.

Reply:
Looks dandy, but to be honest I can't say for sure what's happening under the hood without consulting docs on `Tuple#<=>`, not to mention redundant `Tuple` allocation. IMO it would be more lightweight and understandable if it was written using the more explicit version :)